### ISSUE TITLE
Allow option to tune the prior precision by optimising the NLL on a validation set

### ISSUE THREAD

**Issue by edaxberger** [AlexImmer/Laplace#70]
What do you think about another option for tuning the prior precision that uses (gradient-based) optimization methods to minimise the NLL on a validation set? I think this might nicely complement the existing options (i.e. MLL optimization and CV using validation data).

This is e.g. how the temperature parameter in temperature scaling is typically optimized; see an example implementation using BFGS from scipy [here](https://github.com/JonathanWenger/pycalib/blob/0c023465c5a73d5b90b27e9ff6848a649f45c834/pycalib/calibration_methods.py#L161-L237).

Even easier would be to use the same optimization approach as for the MLL (i.e. Adam from PyTorch).
